src/app/modules/group/components/group-manager-add/group-manager-add.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { GroupData } from '../../services/group-datasource.service';
+import { catchError, switchMap } from 'rxjs/operators';
+import { EMPTY } from 'rxjs';
+import { Manager } from '../../http-services/get-group-managers.service';
+import { GetUserByLoginService } from '../../../../core/http-services/get-user-by-login.service';
+import { GroupCreateManagerService } from '../../http-services/group-create-manager.service';
+import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
+
+@Component({
+  selector: 'alg-group-manager-add',
+  templateUrl: './group-manager-add.component.html',
+  styleUrls: [ './group-manager-add.component.scss' ]
+})
+export class GroupManagerAddComponent {
+
+  @Output() added = new EventEmitter<void>();
+
+  @Input() groupData?: GroupData;
+  @Input() managers?: Manager[];
+
+  state: 'ready' | 'error' | 'loading' = 'ready';
+  login = '';
+
+  constructor(
+    private getUserByLoginService: GetUserByLoginService,
+    private groupCreateManagerService: GroupCreateManagerService,
+    private actionFeedbackService: ActionFeedbackService,
+  ) {}
+
+  onClick(): void {
+    if (!this.managers) {
+      throw new Error('Unexpected: Missed managers');
+    }
+
+    if (this.managers.some(manager => manager.login === this.login)) {
+      this.actionFeedbackService.error($localize`This user is already a manager of this group.`);
+      return;
+    }
+
+    if (!this.groupData) {
+      throw new Error('Unexpected: missed group data');
+    }
+
+    const groupId = this.groupData.group.id;
+
+    this.state = 'loading';
+    this.getUserByLoginService.get(this.login).pipe(
+      catchError(() => {
+        this.state = 'error';
+        this.actionFeedbackService.error($localize`The login you entered does not exist or is not visible to you.`);
+        return EMPTY;
+      }),
+      switchMap(user => this.groupCreateManagerService.create(groupId, user.groupId)),
+    ).subscribe({
+      next: () => {
+        this.state = 'ready';
+        this.actionFeedbackService.success($localize`Manager added!`);
+        this.login = '';
+        this.added.emit();
+      },
+      error: () => {
+        this.state = 'error';
+        this.actionFeedbackService.error($localize`Unable to add this manager.`);
+      },
+    });
```

Comment:
To me error handling is ok but I'd like the opinion of @smadbe

Reply:
No side effects in Rx operators! So L50-51 are not ok! In case of error in `get`, the switchMap will not be called anyway.
And it is not ok to consider all kind of errors of `get` as "The login you entered does not exist or is not visible to you." ... it may be a server error (5xx). 
To adapt the error message to the service in error, either you find a way to find this info in the error metadata (?) or you write a `catchError` for each call which emit a customized error. 
Maybe @ThomasJuster has better idea?

Reply:
Throw specific pre-declared errors and handle them in the `subscribe({ error: () => {…} })`
ie:
```ts
const loginNotFoundError = new Error('login_not_found')
this.getUserByLoginService.get(this.login).pipe(
  catchError((error) => {
    if (errorIsHTTPNotFound(error)) throw loginNotFoundError
    throw error
  }),
  switchMap(() => this.groupCreateManagerService.create(…)),
  catchError((error) => {
     if (errorIsHTTPForbidden(error)) throw 
  }),
)
```

Reply:
Fixed before I've seen Thomas comment, please take a look